Collections.cs

```diff
@@ -77,10 +77,22 @@ internal static void AddExtraSongData(string hash, string path, string rawSongDa
             return diffData;
         }
 
-        internal static async Task LoadExtraSongDataAsync()
+        internal static void LoadExtraSongData()
         {
-            using var reader = new StreamReader(DataPath);
-            CustomSongsData = JsonConvert.DeserializeObject<ConcurrentDictionary<string, ExtraSongData>?>(await reader.ReadToEndAsync()) ?? new ConcurrentDictionary<string, ExtraSongData>();
+            Task.Run(() =>
+            {
+                try
+                {
+                    using var reader = new JsonTextReader(new StreamReader(DataPath));
+                    var serializer = JsonSerializer.CreateDefault();
+                    CustomSongsData = serializer.Deserialize<ConcurrentDictionary<string, ExtraSongData>?>(reader) ?? new ConcurrentDictionary<string, ExtraSongData>();
+                }
+                catch (Exception ex)
+                {
+                    Logging.Logger.Error($"Error loading extra song data: {ex.Message}");
+                    Logging.Logger.Debug(ex);
```

Comment:
You might wanna change this to `.Error(...)`

Reply:
Yeah, I used the same style as other try-catch blocks in `Loader.cs`, I would've used it otherwise.